### PR TITLE
fix(dashboard): Fix tag input suggestion sorting

### PR DESCRIPTION
**Pull request by desiprisg**
### What changed? Why was the change needed?

Now what you have typed will always show first.
### Screenshots
Screenshot 2024-11-11 at 7 58 45 PM

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/dashboard/src/components/primitives/tag-input.tsx [hunk 1]
```diff
@@ -27,10 +27,16 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
...
+    const newTag = tag.trim();
+    if (newTag === '') {
+      return;
+    }
+
     const newTags = [...tags, tag];
     if (new Set(newTags).size !== newTags.length) {
       return;
     }
+
```
Comment (rifont) [inline apps/dashboard/src/components/primitives/tag-input.tsx]: Accidentally brought across from the other PR

```suggestion
    const newTags = [...tags, tag];
    if (new Set(newTags).size !== newTags.length) {
      return;
    }
```

File: apps/dashboard/src/components/primitives/tag-input.tsx [hunk 2]
```diff
@@ -107,9 +113,7 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
               <CommandGroup>
                 {inputValue !== '' && (
                   <CommandItem
-                    // We can't have duplicate keys in our list so adding a prefix
-                    // here to differentiate this from a possible suggestion value
-                    value={`input-${inputValue}`}
+                    value={`${inputValue}`}
```
Comment (rifont) [inline apps/dashboard/src/components/primitives/tag-input.tsx]: Can we simplify to just the value now?
```suggestion
                    value={inputValue}
```

**Review by rifont — APPROVED** [review]
💯

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/components/primitives/tag-input.tsx (hunk 2 above)
Comment (desiprisg) [reply apps/dashboard/src/components/primitives/tag-input.tsx]: 👁️

File: apps/dashboard/src/components/primitives/tag-input.tsx (hunk 1 above)
Comment (desiprisg) [reply apps/dashboard/src/components/primitives/tag-input.tsx]: Thanks! fixed

**Pull request merged**